store/sqlstore/user_store.go

```diff
@@ -397,7 +397,22 @@ func (us SqlUserStore) GetProfilesInChannel(channelId string, offset int, limit
 	return store.Do(func(result *store.StoreResult) {
 		var users []*model.User
 
-		query := "SELECT Users.* FROM Users, ChannelMembers WHERE ChannelMembers.ChannelId = :ChannelId AND Users.Id = ChannelMembers.UserId ORDER BY Users.Username ASC LIMIT :Limit OFFSET :Offset"
+		query := `
+			SELECT 
+				Users.*
+			FROM Users
+				INNER JOIN ChannelMembers ON Users.Id = ChannelMembers.UserId
+				LEFT JOIN Status  ON Users.Id = Status.UserId
+			WHERE
+				ChannelMembers.ChannelId = :ChannelId
+			ORDER BY CASE Status
+				WHEN 'online' THEN 1
+				WHEN 'away' THEN 2
+				WHEN 'dnd' THEN 4
+				ELSE 3
+			END
+			LIMIT :Limit OFFSET :Offset
+		`
```

Comment:
The query looks fine but I don't think we want to change the ordering for absolutely everything using this function. For the majority of the cases I'm pretty sure we want the old behaviour, isn't it just the one instance on the client where we want to sort by status?

I think you need to add a "sort" query parameter to the `GET /channels/{id}/members` API endpoint that will default to the old query and then use this query if "?sort=status" is present

For example, [the GET /emojis endpoint](https://github.com/mattermost/mattermost-server/blob/master/api4/emoji.go#L85)

Reply:
You'll also need to update the API reference to match the new functionality around here https://github.com/mattermost/mattermost-api-reference/blob/master/v4/source/users.yaml#L64